src/intralinks.rs

```diff
@@ -0,0 +1,933 @@
+use std::collections::{HashMap, HashSet};
+use std::fmt;
+use std::fs::File;
+use std::io::Read;
+use std::path::{Path, PathBuf};
+use syn::export::fmt::Display;
+use syn::export::{Debug, Formatter};
+use syn::Ident;
+use syn::Item;
+
+pub enum IntraLinkError {
+  IOError(std::io::Error),
+  ParseError(syn::Error),
+  NoModuleFile(String, PathBuf),
+}
+
+impl fmt::Display for IntraLinkError {
+  fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+    match self {
+      IntraLinkError::IOError(err) => write!(f, "IO error: {}", err),
+      IntraLinkError::ParseError(err) => write!(f, "NoModuleFile error: {}", err),
+      IntraLinkError::NoModuleFile(module, dir) => write!(
+        f,
+        "Unable to find module file for module {} in directory {:?}",
+        module, dir
+      ),
+    }
+  }
+}
+
+impl From<std::io::Error> for IntraLinkError {
+  fn from(err: std::io::Error) -> Self {
+    IntraLinkError::IOError(err)
+  }
+}
+
+impl From<syn::Error> for IntraLinkError {
+  fn from(err: syn::Error) -> Self {
+    IntraLinkError::ParseError(err)
+  }
+}
+
+fn file_ast<P: AsRef<Path>>(file: P) -> Result<syn::File, IntraLinkError> {
+  let mut file = File::open(file)?;
+  let mut src = String::new();
+
+  file.read_to_string(&mut src)?;
```

Comment:
You can replace all this with:

```suggestion
  let src = fs::read_to_string()?;
```

Reply:
(i.e. you’ll need to import `std::fs`).